torch/csrc/jit/passes/onnx/preclude_inlining.cpp

```diff
@@ -0,0 +1,65 @@
+#include <torch/csrc/jit/passes/onnx/preclude_inlining.h>
+#include <torch/csrc/jit/jit_log.h>
+#include <torch/csrc/jit/passes/onnx/helper.h>
+
+namespace torch {
+namespace jit {
+
+void functionCallSubstitution(Block* block) {
+  for (auto it = block->nodes().begin(), end = block->nodes().end();
+       it != end;) {
+    Node* cur = *it++;
+    if (cur->kind() == prim::CallFunction) {
+      AT_ASSERT(cur->input(0)->node()->kind() == prim::Constant);
+      auto function_constant = cur->input(0)->node();
+      auto fun_type =
+          function_constant->output()->type()->expect<FunctionType>();
+
+      if ((fun_type->function()->qualname().qualifiedName().find(
+               "__torch__.torch.nn.functional") != std::string::npos) &&
+          (fun_type->function()->qualname().qualifiedName().find(
+               "interpolate") != std::string::npos)) {
+        cur->removeInput(0);
+        Node* interpolate_node = block->owningGraph()->create(
+            Symbol::fromQualString("aten::__interpolate"),
+            {cur->inputs()},
+            cur->outputs().size());
+        interpolate_node->output()->copyMetadata(cur->output());
+        interpolate_node->insertAfter(cur);
+        cur->replaceAllUsesWith(interpolate_node);
+        cur->removeAllInputs();
+        cur->destroy();
+      } else {
```

Comment:
You also need to recurse through CallMethod nodes

Reply:
Could you help me understand the use of CallMethod? I assumed this is used for builtin function calls (such as call to forward function) but I don't see that in the exported IR graph for my tests.

Reply:
I think you just want something like:
```
      case prim::CallMethod: {
        const std::string& name = cur->s(attr::name);
        if (auto class_type = cur->input(0)->type()->cast<ClassType>()) {
          auto function = class_type->getMethod(name);
          if (!function->isGraphFunction()) {
            continue;
          }
          functionCallSubstitution(fun_type->function()->graph()->block());
        }
      } break;
```

prim::MethodCall are emitted whenever a method is invoked. If you add another layer of method nesting in `_interpolate_script`:
```
class Module():
    def forward(self, x):
         return self.submodule.forward(x)
```
(where submodule is the other module you're testing) that code path should be exercised

Reply:
Got it. Thanks